### PR TITLE
DEP: deprecate transformer tutorial

### LINKED ISSUE #2895: [BUG] - Dependency Issue in Language Modeling with nn.Transformer and torchtext Tutorial
### Issue and Suggested Fix

Please can this helpful tutorial be updated with the HF `from datasets import load_dataset` and merged into main with the dependency issue workaround:

```
import torch
torch.utils.data.datapipes.utils.common.DILL_AVAILABLE = torch.utils._import_utils.dill_available()
import torchdata
```

### Asset 

https://colab.research.google.com/github/pytorch/tutorials/blob/gh-pages/_downloads/9cf2d4ead514e661e20d2070c9bf7324/transformer_tutorial.ipynb#scrollTo=TY5T9Gic_qih

### Describe the bug

```
ImportError                               Traceback (most recent call last)
[<ipython-input-26-b02c7921f3b1>](https://localhost:8080/#) in <cell line: 5>()
      3 from torchtext.vocab import build_vocab_from_iterator
      4 
----> 5 train_iter = WikiText2(split='train')
      6 tokenizer = get_tokenizer('basic_english')
      7 vocab = build_vocab_from_iterator(map(tokenizer, train_iter), specials=['<unk>'])

6 frames
[/usr/local/lib/python3.10/dist-packages/torchdata/datapipes/iter/util/cacheholder.py](https://localhost:8080/#) in <module>
     22     portalocker = None
     23 
---> 24 from torch.utils.data.datapipes.utils.common import _check_unpickable_fn, DILL_AVAILABLE
     25 
     26 from torch.utils.data.graph import traverse_dps

ImportError: cannot import name 'DILL_AVAILABLE' from 'torch.utils.data.datapipes.utils.common' (/usr/local/lib/python3.10/dist-packages/torch/utils/data/datapipes/utils/common.py)

---------------------------------------------------------------------------
NOTE: If your import is failing due to a missing package, you can
manually install dependencies using either !pip or !apt.

To view examples of installing some common dependencies, click the
"Open Examples" button below.
---------------------------------------------------------------------------
```

### Describe your environment

Google Colab environment. Have replicated the issue locally with same pip package versions.

cc @sekyondaMeta @svekars @kit1980

### PR DESCRIPTION
**Pull request by loganthomas**
Fixes #2895 

## Checklist

- [X] The issue that is being fixed is referred in the description (see above "Fixes #ISSUE_NUMBER")
- [X] Only one issue is addressed in this pull request
- [X] Labels from the issue that this PR is fixing are added to this pull request
- [X] No unnecessary issues are included into this pull request.

cc @sekyondaMeta @svekars @kit1980 @brycebortree

### CODE REVIEW

**Comment by loganthomas** [comment]
@svekars
It looks like there is another advanced tutorial that has the same issue:
- [Source code permalink](https://github.com/pytorch/tutorials/blob/653719940f7c4d908811da415f190465d8c3189d/advanced_source/ddp_pipeline.py#L175)
- [Online docs link](https://pytorch.org/tutorials/advanced/ddp_pipeline.html)
- [Colab link](https://colab.research.google.com/github/pytorch/tutorials/blob/gh-pages/_downloads/9c42ef95b5e306580f45ed7f652191bf/ddp_pipeline.ipynb#scrollTo=E2Lgxp7SSfUS)

Shall I deprecate this one as well in this PR? Or, submit another issue/PR to resolve that one?

**Comment by svekars** [comment]
> @svekars It looks like there is another advanced tutorial that has the same issue:
> 
> * [Source code permalink](https://github.com/pytorch/tutorials/blob/653719940f7c4d908811da415f190465d8c3189d/advanced_source/ddp_pipeline.py#L175)
> * [Online docs link](https://pytorch.org/tutorials/advanced/ddp_pipeline.html)
> * [Colab link](https://colab.research.google.com/github/pytorch/tutorials/blob/gh-pages/_downloads/9c42ef95b5e306580f45ed7f652191bf/ddp_pipeline.ipynb#scrollTo=E2Lgxp7SSfUS)
> 
> Shall I deprecate this one as well in this PR? Or, submit another issue/PR to resolve that one?

Can you please create a separate task for this work? We will need to work with the owner of this tutorial to understand if there is any other solutions they want to propose.

File: beginner_source/transformer_tutorial.rst [hunk 1]
```diff
@@ -0,0 +1,7 @@
+Language Modeling with nn.Transformer and torchtext
```
Comment (svekars) [inline beginner_source/transformer_tutorial.rst:1]:
```suggestion
Language Modeling with ``nn.Transformer`` and torchtext
```

File: beginner_source/transformer_tutorial.rst [hunk 2]
```diff
@@ -0,0 +1,7 @@
+Language Modeling with ``nn.Transformer`` and torchtext
+==========================================
```
Comment (svekars) [inline beginner_source/transformer_tutorial.rst:2]:
```suggestion
===========================================================
```

**Review by svekars — APPROVED** [review]

**Comment by loganthomas** [comment]
> > @svekars It looks like there is another advanced tutorial that has the same issue:
> > 
> > * [Source code permalink](https://github.com/pytorch/tutorials/blob/653719940f7c4d908811da415f190465d8c3189d/advanced_source/ddp_pipeline.py#L175)
> > * [Online docs link](https://pytorch.org/tutorials/advanced/ddp_pipeline.html)
> > * [Colab link](https://colab.research.google.com/github/pytorch/tutorials/blob/gh-pages/_downloads/9c42ef95b5e306580f45ed7f652191bf/ddp_pipeline.ipynb#scrollTo=E2Lgxp7SSfUS)
> > 
> > Shall I deprecate this one as well in this PR? Or, submit another issue/PR to resolve that one?
> 
> Can you please create a separate task for this work? We will need to work with the owner of this tutorial to understand if there is any other solutions they want to propose.

See # 2916

**Pull request merged**